File: stacks/rails/app/controllers/search_controller.rb
# 検索（spec/screens.md #4）。飼主・動物と、診察の中身を独立に検索する。
class SearchController < ApplicationController
  def index
    @q = params[:q].to_s.strip
    if @q.present?
      @patients = Patient.kept.includes(:owner)
                          .where("name_kanji LIKE :q OR name_kana LIKE :q OR karte_no LIKE :q",
                                 q: "%#{@q}%")
                          .limit(20)
      @visits = Visit.kept.includes(:patient)
                     .where("chief_complaint LIKE :q OR symptom LIKE :q OR diagnosis LIKE :q OR treatment LIKE :q",
                            q: "%#{@q}%")
                     .limit(20)
    else
      @patients = []
      @visits = []
    end
  end
end

File: stacks/laravel/app/Http/Controllers/Reception/SearchController.php
<?php

namespace App\Http\Controllers\Reception;

use App\Http\Controllers\Controller;
use App\Models\Patient;
use Illuminate\Contracts\View\View;
use Illuminate\Http\Request;

/** 検索。契約は spec/openapi.yaml `/search`。飼主名・カナ・カルテNo・電話番号・診療の中身を横断する。 */
class SearchController extends Controller
{
    public function index(Request $request): View
    {
        $q = trim((string) $request->query('q', ''));
        $patients = collect();

        if ($q !== '') {
            $patients = Patient::with('owner')
                ->where('karte_no', 'like', "%{$q}%")
                ->orWhere('name_kanji', 'like', "%{$q}%")
                ->orWhere('name_kana', 'like', "%{$q}%")
                ->limit(50)
                ->get();
        }

        return view('reception.search', ['q' => $q, 'patients' => $patients]);
    }
}
